src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -86491,5 +86491,52 @@ public void M(TypedReference r)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "__refvalue(r, string?)").WithLocation(7, 13)
                 );
         }
+
+        [WorkItem(30987, "https://github.com/dotnet/roslyn/issues/30987")]
+        [Fact]
+        public void TypeInference_LowerBounds_TopLevelNullability_03()
```

Comment:
Can you add some tests that demonstrate what happens when you involve a parameter defined in a null-disabled context in the mix?

Reply:
If you mean parameters of `G`, those just read as non-nullable.  If you mean parameters of `F`, it shouldn't make a difference.  However, I'll add tests.

---
In reply to: [267929254](https://github.com/dotnet/roslyn/pull/34270#discussion_r267929254) [](ancestors = 267929254)